Log modules FSDP hooks fire for

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #102890
* #102692
* #102680
* #102672
* #102658
* #102657
* #102639
* #102637
* __->__ #102508

Under torch_distributed_debug >= INFO and use_orig_params=True, log post backward hook firing to debug things like FSDP + AC integration.

Differential Revision: [D46172916](https://our.internmc.facebook.com/intern/diff/D46172916/)

rohan-varma committed:
Log modules FSDP hooks fire for

Under torch_distributed_debug >= INFO and use_orig_params=True, log post backward hook firing to debug things like FSDP + AC integration.

Differential Revision: [D46172916](https://our.internmc.facebook.com/intern/diff/D46172916/)

[ghstack-poisoned]

awgu commented on torch/distributed/fsdp/_runtime_utils.py:
```diff
@@ -735,6 +736,19 @@ def _post_backward_hook(
     - Otherwise, the ``_saved_grad_shard`` attribute is the reduced sharded
     gradient (accumulating with any existing gradient).
     """
```
`FlatParamHandle` has an attribute:
https://github.com/pytorch/pytorch/blob/0f1621df1a0a73956c7ce4e2f72f069e610e0137/torch/distributed/fsdp/flat_param.py#L499

awgu commented on torch/distributed/fsdp/_runtime_utils.py:
```diff
@@ -735,6 +736,19 @@ def _post_backward_hook(
+            for param_list in [param_to_fqn[p] for p in handle_params]
+            for param in param_list
+        ]
+        logging.warning(f"FSDP firing hooks for parameters {param_fqns}")
```
We should say that this is the post-backward hook.

awgu commented on torch/distributed/fsdp/_runtime_utils.py:
```diff
@@ -735,6 +736,19 @@ def _post_backward_hook(
     - Otherwise, the ``_saved_grad_shard`` attribute is the reduced sharded
     gradient (accumulating with any existing gradient).
     """
+    # Under TORCH_DISTRIBUTED_DEBUG=INFO, log the module names this hook fires for.
```
nit: IMHO, only the part that explains when we expect hooks to not fire (FSDP + AC) is important info the document. The other info is all clear from the code (e.g. line 742):
```
if state._use_orig_params and dist.get_debug_level() == dist.DebugLevel.INFO:
```

rohan-varma committed:
Update on "Log modules FSDP hooks fire for"


Under torch_distributed_debug >= INFO and use_orig_params=True, log post backward hook firing to debug things like FSDP + AC integration.

Differential Revision: [D46172916](https://our.internmc.facebook.com/intern/diff/D46172916/)

[ghstack-poisoned]

awgu reviewed (approved):
Does the post-backward hook only not fire for reentrant AC?

rohan-varma committed:
Update on "Log modules FSDP hooks fire for"


Under torch_distributed_debug >= INFO and use_orig_params=True, log post backward hook firing to debug things like FSDP + AC integration.

Differential Revision: [D46172916](https://our.internmc.facebook.com/intern/diff/D46172916/)

[ghstack-poisoned]

rohan-varma committed:
Update on "Log modules FSDP hooks fire for"


Under torch_distributed_debug >= INFO and use_orig_params=True, log post backward hook firing to debug things like FSDP + AC integration.

Differential Revision: [D46172916](https://our.internmc.facebook.com/intern/diff/D46172916/)

[ghstack-poisoned]

rohan-varma committed:
Update on "Log modules FSDP hooks fire for"


Under torch_distributed_debug >= INFO and use_orig_params=True, log post backward hook firing to debug things like FSDP + AC integration.

Differential Revision: [D46172916](https://our.internmc.facebook.com/intern/diff/D46172916/)

[ghstack-poisoned]
